haystack/components/connectors/openapi_service.py

```diff
@@ -0,0 +1,145 @@
+        return {"service_response": [ChatMessage.from_user(str(service_response))]}
+
+    def _parse_message(self, content: str) -> Dict[str, Any]:
+        """
+        Parses the message content to extract the method invocation descriptor.
+
+        :param content: The JSON string content of the message.
+        :type content: str
+        :return: A dictionary with method name and arguments.
+        :rtype: Dict[str, Any]
+        :raises ValueError: If the content is not valid JSON or lacks required fields.
+        """
+        if not self._is_valid_json(content):
+            raise ValueError("Invalid JSON content, cannot parse invocation message.", content)
+
+        descriptor = json.loads(content)
+        if "name" not in descriptor or "arguments" not in descriptor:
+            raise ValueError("Missing required fields in the invocation message content.", content)
+
+        descriptor["arguments"] = json.loads(descriptor["arguments"])
+        return descriptor
+
+    def _authenticate_service(self, openapi_service: OpenAPI):
+        """
+        Authenticates with the OpenAPI service if required.
+
+        :param openapi_service: The OpenAPI service instance.
+        :type openapi_service: OpenAPI
+        :raises ValueError: If authentication fails or is not found.
+        """
+        if openapi_service.components.securitySchemes:
+            auth_method = list(openapi_service.components.securitySchemes.keys())[0]
+            service_title = openapi_service.info.title
+            if service_title not in self.service_authentications:
+                raise ValueError(f"Service {service_title} not found in service_authentications.")
+            openapi_service.authenticate(auth_method, self.service_authentications[service_title])
+
+    def _invoke_method(self, openapi_service: OpenAPI, method_invocation_descriptor: Dict[str, Any]) -> Any:
+        """
+        Invokes the specified method on the OpenAPI service.
+
+        :param openapi_service: The OpenAPI service instance.
+        :type openapi_service: OpenAPI
+        :param method_invocation_descriptor: The method name and arguments.
+        :type method_invocation_descriptor: Dict[str, Any]
+        :return: A service JSON response.
+        :rtype: Any
+        :raises RuntimeError: If the method is not found or invocation fails.
+        """
+        name = method_invocation_descriptor["name"]
+        # a bit convoluted, but we need to pass parameters, data, or both to the method
+        # depending on the openapi operation specification, can't use None as a default value
+        method_call_params = {
+            key: value
+            for key, value in {
+                "parameters": method_invocation_descriptor["arguments"].get("parameters"),
+                "data": method_invocation_descriptor["arguments"].get("requestBody"),
+            }.items()
+            if value is not None
+        }
```

Comment:
What do you think of this? Seems a bit more readable to me. 🤔 

```suggestion
        method_call_params = {}
        if (parameters := method_invocation_descriptor["arguments"].get("parameters")) is not None:
            method_call_params["parameters"] = parameters
        if (arguments := method_invocation_descriptor["arguments"].get("requestBody")) is not None:
            method_call_params["data"] = arguments
```